### PR TITLE
Logging: add logging for moveToTrash

### LINKED ISSUE #1106: Could not delete image: ...
**Describe the bug**
When I press delete a dialog box appears on the bottom left saying "Could not delete image: [filename]".

**To Reproduce**
1. Open an image with nomacs
2. Press delete
3. See error

**Expected behavior**
No error should be present, or at the very least, there should be a more informative log message generated on why this failure occurred.

**Screenshots**
<video src="https://github.com/nomacs/nomacs/assets/20902250/b5ece8b8-b6b6-4997-ae31-b0a5b5b3acd5">

**Desktop (please complete the following information):**
 - OS: NixOS, nixpkgs rev 051f920625ab5aabe37c920346e3e69d7d34400e
 - Nomacs version: Image Lounge 3.17.0 

**Additional context**
I have enabled logging and this is the log file until the moment I pressed delete and that text appeared. There doesn't seem to be any indication of why this failure occurred:

```
Hi there
Metadata support for BMFF formats is active.
plugins dir set to:  "/nix/store/qpk85xnnx7w7fc8zkiq6mav3n4jiiffr-qtbase-5.15.12/lib/nomacs-plugins"
local client created in:  2 ms
theme loaded from "/nix/store/b1kpf5vhwfb4gqb2jrl06zb4agqvggfr-nomacs-3.17.2295/share/nomacs/Image Lounge/themes/Dark-Theme.css"
CSS loaded from:  ":/nomacs/stylesheet.css"
Initialization takes:  96 ms
[DkImageLoader] 40 containers created in 1 ms
[DkImageLoader] after sorting:  1 ms
[Basic Loader] "/.btrfs/old/@home/itsdrike/Pictures/Screenshots/Screenshot_2024-01-07_23-13-29.png" loaded in 13 ms
[Basic Loader] "/.btrfs/old/@home/itsdrike/Pictures/Screenshots/Screenshot_2024-01-16_10-22-57.png" loaded in 5 ms
```

Deleting the image manually using `trash-put [image path]` works without issues.

### PR DESCRIPTION
**Pull request by scrubbbbs**
There are definitely some cases where QFile::moveToTrash() will fail, this adds some logging that might help solve them.

~Fixes #1106~

### CODE REVIEW

File: ImageLounge/src/DkCore/DkUtils.cpp [hunk 1]
```diff
@@ -842,18 +842,34 @@ bool DkUtils::moveToTrash(const QString &filePath)
 
     // delete links
     if (fileInfo.isSymLink()) {
+        qInfo() << "move to trash: deleting symlink"  << filePath;
         QFile fh(filePath);
         return fh.remove();
```
Comment (leejuyuu) [inline ImageLounge/src/DkCore/DkUtils.cpp:847]: Maybe also add another warning when this `remove()` call fails?

**Comment by leejuyuu** [comment]
I think this should not close https://github.com/nomacs/nomacs/issues/1106.

**Comment by scrubbbbs** [comment]
@leejuyuu  Ok, maybe we should keep it open

- The added logging addresses the feature/enhancement
- The bug is with Qt or system configuration, however we could have a fallback like using `trash-put` on Unix or using the Windows/Mac trash APIs directly.

**Comment by leejuyuu** [comment]
Until we can pinpoint the source of the issue (which this logging PR will help a lot), the issue should stay open. Even if it is an upstream bug or missing feature, it should be open to track whether the upstream fixed it.

**Pull request merged**